unstructured_ingest/v2/processes/connectors/pinecone.py

```diff
@@ -28,6 +28,7 @@
 CONNECTOR_TYPE = "pinecone"
 MAX_PAYLOAD_SIZE = 2 * 1024 * 1024  # 2MB
```

Comment:
Metadata has its own limit per record, so we have to check it separately anyway.
There's hard 40KB limit on just metadata, to quote from [docs](https://docs.pinecone.io/reference/quotas-and-limits#hard-limits)

> Max metadata size per record is 40 KB.

Instead of cropping the metadata I can drop all of it but the new size check is still required.

Reply:
Ah nice catch, I didn't realize there were different size limitations. If the size is over, lets drop all of the metadata then.